cmd/deploy.go

```diff
@@ -404,6 +404,12 @@ func (dc *deployCmd) run() error {
 		return errors.Wrapf(err, "in SetPropertiesDefaults template %s", dc.apimodelPath)
 	}
 
+	// Validate image
+	err = dc.validateDependencies(context.Background())
```

Comment:
`context.Background()` is not appropriate to use since it means that this command could wait indefinitely for validateDependencies to return.

You should create a context with the maximal amount of time you expect this call to take.